thorlcr/graph/thgraphslave.cpp

```diff
@@ -540,16 +540,23 @@ unsigned __int64 CSlaveActivity::queryLocalCycles() const
                 break;
         }
     }
-    unsigned __int64 _totalCycles = queryTotalCycles();
-    if (_totalCycles < inputCycles) // not sure how/if possible, but guard against
+    const unsigned __int64 _totalCycles = queryTotalCycles();
+    const unsigned __int64 blockedCycles = queryBlockedCycles();
+    if (_totalCycles < inputCycles || blockedCycles > (_totalCycles-inputCycles)) // not sure how/if possible, but guard against
```

Comment:
minor: but as this should not be the norm. would be slightly better to avoid this arithmetic twice, e.g. could be:
```
    unsigned __int64 localCycles = queryTotalCycles();
    if (localCycles < inputCycles)
        return 0;
    localCycles -= inputCycles;
    const unsigned __int64 blockedCycles = queryBlockedCycles();
    if (localCycles < blockedCycles) // not sure how/if possible, but guard against
        return 0;
    return localCycles-blockedCycles;
```